g3doc/ConversionToLLVMDialect.md

```diff
@@ -90,6 +90,24 @@ memref<10x?x42x?x123 x f32> -> !llvm.type<"{ float*, float*, i64, [5 x i64], [5
 memref<1x? x vector<4xf32>> -> !llvm.type<"{ <4 x float>*, <4 x float>*, i64, [1 x i64], [1 x i64] }">
 ```
 
+If the rank of the memref is unknown at compile time, the Memref is converted to an unranked 
+descriptor that contains:
+1. a 64-bit integer representing the dynamic rank of the memref, followed by
+2. a pointer to a ranked memref descriptor with the contents listed above. 
+
+Dynamic ranked memrefs should be used only to pass arguments to external library calls
+that expect a unified memref type. The called functions can parse any unranked memref
+descriptor by reading the rank and parsing the enclosed ranked descriptor.
+
+Examples:
+
+```mlir {.mlir}
+// unranked descriptor containing a 0-D memref
+memref<*xf32> -> !llvm.type<"{i64,{ float*, float*, i64 }*}">  
+// unranked descriptor containing a 2-D memref
+memref<*xf32> -> !llvm.type<"{i64,{ float*, float*, i64, [2 x i64], [2 x i64] }*}">  
```

Comment:
Your PR description says:
```
The LLVM descriptor consists of a runtime dynamic rank and a pointer to ranked MemRef llvm<"{ i64, i8* }
```

The only way you can know that it contains a 2-D memref is by looking at the cast op which has non-local implications.
In particular `i8*` is the only local information you can derive looking at the function signature.

I think `i8*` is better and let the library inspect the `i64` and cast the `void*` into whatever.

Any particular reason you went contrary to what your PR message originally described?

Reply:
My bad, just an oversight. Fixed now. Thanks for catching this.